core/src/event_id.rs

```diff
@@ -149,9 +118,211 @@ fn last4_bytes(buf: &[u8]) -> &[u8] {
     &buf[(buf.len() - 4)..]
 }
 
+const ZEROS_8: &[u8] = &[0; 8];
+const FFS_8: &[u8] = &[0xFF; 8];
+const ZEROS_4: &[u8] = &[0; 4];
+const FFS_4: &[u8] = &[0xFF; 4];
+
+/// Builder provides an ordered API for constructing an EventId
+#[derive(Debug)]
+pub struct Builder<S: BuilderState> {
+    state: S,
+}
+/// The state of the builder
+pub trait BuilderState {}
+
+/// Initial state of the builder.
+#[derive(Debug)]
+pub struct Init;
+impl BuilderState for Init {}
+
+/// Builder with network set.
+#[derive(Debug)]
+pub struct WithNetwork {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithNetwork {}
+
+/// Builder with sort value set.
+#[derive(Debug)]
+pub struct WithSortValue {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithSortValue {}
+
+/// Builder with controller set.
+#[derive(Debug)]
+pub struct WithController {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithController {}
+
+/// Builder with init event CID set.
+#[derive(Debug)]
+pub struct WithInit {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithInit {}
+
+/// Builder with event height set.
+#[derive(Debug)]
+pub struct WithEventHeight {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithEventHeight {}
+
+/// Builder with event CID set.
+#[derive(Debug)]
+pub struct WithEvent {
+    bytes: Vec<u8>,
+}
+impl BuilderState for WithEvent {}
+
+impl Builder<Init> {
+    pub fn with_network(self, network: &Network) -> Builder<WithNetwork> {
+        // TODO what is max event id size?
```

Comment:
varint(0xce) + // streamid, `1` byte
varint(0x05) + // cip-124 EventID, `1` byte
varint(networkId), // `5` bytes for local network
last8Bytes(sha256(separator_value)), // `8` bytes
last8Bytes(sha256(stream_controller_DID)), // `8` bytes
last4Bytes(init_event_CID) // `4` bytes
cbor(eventHeight), // u64_max `9` bytes
eventCID ? mostly `36` bytes but could be inline CID 

`1 + 1 + 5 + 8 + 8 + 4 + 9 + ?`
`36 + ?`
assuming 36 bytes for 32 byte hash CID
`36 + 36`
`72`